x/quarantine/spec/06_client.md

```diff
@@ -0,0 +1,222 @@
+
+Usage:
+  simd tx quarantine opt-out [<to_name_or_address>] [flags]
+
+Examples:
+
+$ simd tx quarantine opt-out cosmos1c7p4v02eayvag8nswm4f5q664twfe6dxjha389
+$ simd tx quarantine opt-out personal
+$ simd tx quarantine opt-out --from cosmos1c7p4v02eayvag8nswm4f5q664twfe6dxjha389
+$ simd tx quarantine opt-out --from personal
+```
+
+#### Accept
+
+```shell
+$ ./build/simd tx quarantine accept --help
+Accept quarantined funds sent to <to_name_or_address> from <from_address>.
+Note, the '--from' flag is ignored as it is implied from [to_name_or_address] (the signer of the message).
+
+Usage:
+  simd tx quarantine accept <to_name_or_address> <from_address> [<from_address 2> ...] [flags]
+
+Examples:
+
+$ simd tx quarantine accept cosmos1c7p4v02eayvag8nswm4f5q664twfe6dxjha389 cosmos1ld2qyt9pq5n8dxkp58jn3jyxh8u8ztmrk9vrut
+$ simd tx quarantine accept personal cosmos1ld2qyt9pq5n8dxkp58jn3jyxh8u8ztmrk9vrut
+$ simd tx quarantine accept personal cosmos1ld2qyt9pq5n8dxkp58jn3jyxh8u8ztmrk9vrut cosmos1phx24ecmuw3s7fmy8c87gh3rdq5lwskqur3t00
+```
+
+At least one `<from_address>` is required, but multiple can be provided.
+
+A `--permanent` flag is also available with this command:
+
+```shell
+      --permanent                Also set auto-accept for sends from any of the from_addresses to to_address
+```
+
+#### Decline
+
+```shell
+$ simd tx quarantine decline --help
+Decline quarantined funds sent to <to_name_or_address> from <from_address>.
+Note, the '--from' flag is ignored as it is implied from [to_name_or_address] (the signer of the message).
+
+Usage:
+  simd tx quarantine decline <to_name_or_address> <from_address> [<from_address 2> ...] [flags]
+
+Examples:
+
+$ simd tx quarantine decline cosmos1c7p4v02eayvag8nswm4f5q664twfe6dxjha389 cosmos1ld2qyt9pq5n8dxkp58jn3jyxh8u8ztmrk9vrut
+$ simd tx quarantine decline personal cosmos1ld2qyt9pq5n8dxkp58jn3jyxh8u8ztmrk9vrut
+$ simd tx quarantine decline personal cosmos1ld2qyt9pq5n8dxkp58jn3jyxh8u8ztmrk9vrut cosmos1phx24ecmuw3s7fmy8c87gh3rdq5lwskqur3t00
+```
+
+At least one `<from_address>` is required, but multiple can be provided.
+
+A `--permanent` flag is also available with this command:
+
+```shell
+      --permanent                Also set auto-decline for sends from any of the from_addresses to to_address
+```
+
+#### UpdateAutoResponses
+
+```shell
+$ simd tx quarantine update-auto-responses --help
+Update auto-responses for transfers to <to_name_or_address> from one or more addresses.
+Note, the '--from' flag is ignored as it is implied from [to_name_or_address] (the signer of the message).
+
+The <to_name_or_address> is required.
+At least one <auto-response> and <from_address> must be provided.
+
+Valid <auto-response> values:
+  "accept" or "a" - turn on auto-accept for the following <from_address>(es).
+  "decline" or "d" - turn on auto-decline for the following <from_address>(es).
+  "unspecified", "u", "off", or "o" - turn off auto-responses for the following <from_address>(es).
```

Comment:
This should just be `off`. 

```suggestion
 "off", or "o" - turn off auto-responses for the following <from_address>(es).
```

Reply:
Since the enum value is `unspecified`, that should be an option here too.

Reply:
I understand why the guides suggest the suffix and it makes sense at the proto message level since that is the default value for unexpected unum values but do we have to expose `unspecified` to the end user? It's not a good UX IMO. We can stick with the three states `accept`, `decline` and `off`, they're clear and concise.

> The zero value enum should have the suffix UNSPECIFIED, because a server or application that gets an unexpected enum value will mark the field as unset in the proto instance. The field accessor will then return the default value, which for enum fields is the first enum value.

Reply:
Since the enum field is named `unspecified`, and these strings correlate with auto-response enum entries, it needs to be an option as a string.

If we take out `unspecified` and `u`, leaving only `off` and `o`:
* It's not as obvious what `off` or `o` translates to.
* If you know you want an `AUTO_RESPONSE_UNSPECIFIED`, it's clear what you need to do.

By calling a thing the same in all places, it makes it easier to use. Honestly, I was torn on allowing `off` or `o` here, but decided to include them due to the nicer semantics.

Why do you think it's not good UX?

Would it be enough to just reorder them like this:
```suggestion
  "off", "o", "unspecified", or "u",  - turn off auto-responses for the following <from_address>(es).
```

Reply:
> Why do you think it's not good UX?
>
> Would it be enough to just reorder them like this:

It's not the reordering I'm concerned with, it's the UX. `unspecified` represents an initial state. Meaning the user hasn't not yet made a selection for this service. What happens when the user `opt-in` to `accept` or `decline` and then wants to disable auto-responses? The service is no longer in an `unspecified` state, it is in an `off` state. Also, the description reads `turn off auto-responses for the following <from_address>(es)`

> By calling a thing the same in all places, it makes it easier to use. Honestly, I was torn on allowing off or o here, but decided to include them due to the nicer semantics.

We can more easily define the auto-response service with three states; `accept`, `decline` and `off - default`. It makes for a better UX and nicer semantics (we're all quite familiar to `off`). 

What about creating middle ground between UX and proto guidelines by mapping  ` val AUTO_RESPONSE_OFF = AUTO_RESPONSE_UNSPECIFIED` in the code?

If none of the above doesn't sound right that consider using only `unspecified` and remove `off`. `"unspecified", or "u",  - no auto-response action selected for the following <from_address>(es).` or something like that.

Reply:
In CLI, the easier it is to guess the words and phrases of a command, the easier it is to use the command. The `unspecified` option should be available since that's what the enum entry is named, so will sometimes be what people expect to use. The most common driving thought, though, will be, "I need to turn off an auto-response," so it makes sense to have 'off' be an option too.

> What about creating middle ground between UX and proto guidelines by mapping  `val AUTO_RESPONSE_OFF = AUTO_RESPONSE_UNSPECIFIED` in the code?

The protos drive everything. They're the root source of API documentation, and generating code from them is the first step in using our blockchain. The CLI stuff, specially `tx` and `query` pieces, are just wrappers on the `gRPC` calls defined by the protos. Defining that alias would only affect the `go` code, though, leading to one name used in _our_ `go` code, and another used everywhere else.

> What happens when the user `opt-in` to `accept` or `decline` and then wants to disable auto-responses?

Opting in or out is separate from auto-responses. A user opts themselves into quarantine, then they can set up auto-responses for funds coming from specific other addresses if they want to. When they opt in, an _opt-in_ record is created for the receiver. When they define an auto-response action to take from a sender, an _auto-response_ record is created indicating the receiver, sender, and whether it's auto-accept or auto-decline. To disable the auto-response to a receiver from a sender, that _auto-response_ record is deleted.

> `unspecified` represents an initial state.

This isn't quite accurate. Yes, an initial state is `unspecified`, but `unspecified` does not mean the thing is in an initial state; things can go back to `unspecified`.

> The service is no longer in an `unspecified` state.

When setting an auto-response to `unspecified` or `off`, it deletes the appropriate auto-response record, so it actually does go back to an `unspecified` state.

---

I feel like the primary driving thought that will lead to use of this option in the command is, "I need to turn off my auto-responses from an address." From there, there's two paths someone can take:

1. They start with the CLI, looking at `provenanced tx quarantine update-auto-responses --help`. They will see that `off` is an option, and use that in the command they end up running.
2. They look up the protos to see how to update auto-responses. That would lead them to the `UpdateAutoResponses` tx endpoint which involves a list of `AutoResponseUpdate`s. That message has a comment on the `response` field indicating that `AUTO_RESPONSE_UNSPECIFIED` is what they want. From there, they might use the CLI, but they might use something else. Either way, they are now looking to use `unspecified` now.

---

The `AutoResponseUpdate`s message has this:

```
  // response is the automatic action to take on funds sent from from_address, or in the
  // case of AUTO_RESPONSE_UNSPECIFIED, that no automatic action should be taken.
  AutoResponse response = 2;
```

I'm going to update that to include the word "off" in the description so that there's a bit more parity between the proto and CLI. I will also update the comment on the enum entry to include the word "off".

Reply:
👏 The comments are much better now. 

> 1. They start with the CLI, looking at provenanced tx quarantine update-auto-responses --help. They will see that off is an option, and use that in the command they end up running.

This is why I think that `unspecified` doesn't really need to be an option on the CLI`. The choice almost always will be `off`. We can add comments to the `--help` and specify that off maps to  AUTO_RESPONSE_UNSPECIFIED in the protos.